### PR TITLE
don't require application element and project ID until deploy

### LINKED ISSUE #522: don't include the application element instead of making it empty
If the user does not supply a project ID in the new application wizard, don't include the application element instead of making it empty:

&lt;appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
  &lt;application>&lt;/application>
  &lt;version>1&lt;/version>
...

This may be made obsolete depending on the answer to #520

### PR DESCRIPTION
**Pull request by elharo**
@chanseokoh fix #522

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplatesTest.java [hunk 1]
```diff
@@ -92,10 +91,14 @@ public void testMaterialize()
...
-    NodeList applicationElements = doc.getDocumentElement().getElementsByTagName("application");
-    Assert.assertEquals("Must have exactly one application", 1, applicationElements.getLength());
-    String projectId = applicationElements.item(0).getTextContent();
-    Assert.assertEquals("", projectId);
+    NodeList threadsafeElements = doc.getDocumentElement().getElementsByTagName("threadsafe");
+    Assert.assertEquals("Must have exactly one threadsafe", 1, threadsafeElements.getLength());
+    String threadsafe = threadsafeElements.item(0).getTextContent();
+    Assert.assertEquals("true", threadsafe);
+    NodeList sessionsEnabledElements = doc.getDocumentElement().getElementsByTagName("sessions-enabled");
+    Assert.assertEquals("Must have exactly one threadsafe", 1, sessionsEnabledElements.getLength());
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplatesTest.java]: `"Must have exactly one sessions-enabled"`

**Comment by elharo** [comment]
Maven's a separate issue that needs to be addressed in the archetypes.

**Comment by chanseokoh** [comment]
I mean the Maven wizard screen still has the project id field:

maven-based app engine standard project _008

**Comment by chanseokoh** [comment]
If you decide to remove the project ID from the Maven wizard, I think you can also remove `PreferenceUtil.java` completely. It's used only by the wizard.

**Comment by chanseokoh** [comment]
Need to fix SWT bot tests too:

```
testWithPackageAndProjectId(com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest)  Time elapsed: 10.778 sec  <<< ERROR!
org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException: Could not find widget matching: (of type 'Text' and with label (with mnemonic 'App Engine Project ID: (optional)'))
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:437)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:411)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntil(SWTBotFactory.java:399)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.waitUntilWidgetAppears(SWTBotFactory.java:381)
	at org.eclipse.swtbot.swt.finder.SWTBotFactory.widget(SWTBotFactory.java:334)
	at org.eclipse.swtbot.swt.finder.SWTBot.textWithLabel(SWTBot.java:1327)
	at org.eclipse.swtbot.swt.finder.SWTBot.textWithLabel(SWTBot.java:1315)
	at com.google.cloud.tools.eclipse.integration.appengine.SwtBotAppEngineActions.createNativeWebAppProject(SwtBotAppEngineActions.java:69)
	at com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest.createAndCheck(NewNativeAppEngineStandardProjectTest.java:74)
	at com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest.testWithPackageAndProjectId(NewNativeAppEngineStandardProjectTest.java:66)
```

**Comment by elharo** [comment]
PTAL @chanseokoh

**Comment by elharo** [comment]
Maven wizard changes require changes to the archetypes first.

**Pull request merged**